From 1c9b9f567d1f542cca290dd37993817bac399f0b Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sujeet.baranwal@cavium.com>
Date: Tue, 27 Nov 2018 04:01:12 +0300
Subject: [PATCH 0783/1051] mmc: octeontx2: use device tree entries

The characterstic of the device shall be determined
based on the parameters mentioned in the device tree
instead of hardcoding in the driver.

Signed-off-by: Sujeet Baranwal <sujeet.baranwal@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index 5a1b6dcbb9ac..2d2eee419509 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -151,7 +151,7 @@ static int cvm_mmc_configure_delay(struct cvm_mmc_slot *slot)
 	u64 timing = 0, emm_sample;
 	struct cvm_mmc_host *host = slot->host;
 
-	if (is_mmc_otx2(host)) {
+	if (!is_mmc_8xxx(host)) {
 		/* SDR, data out delay is zero */
 		slot->data_out_tap = 0;
 
@@ -1222,15 +1222,20 @@ int cvm_mmc_of_slot_probe(struct device *dev, struct cvm_mmc_host *host)
 	mmc->ops = &cvm_mmc_ops;
 
 	/*
-	 * We only have a 3.3v supply, we cannot support any
-	 * of the UHS modes. We do support the high speed DDR
-	 * modes up to 52MHz.
+	 * We only have a 3.3v supply for slots, we cannot
+	 * support any of the UHS modes. We do support the
+	 * high speed DDR modes up to 52MHz.
 	 *
 	 * Disable bounce buffers for max_segs = 1
 	 */
+
 	mmc->caps |= MMC_CAP_MMC_HIGHSPEED | MMC_CAP_SD_HIGHSPEED |
-		     MMC_CAP_ERASE | MMC_CAP_CMD23 | MMC_CAP_POWER_OFF_CARD |
-		     MMC_CAP_3_3V_DDR;
+		MMC_CAP_ERASE | MMC_CAP_CMD23 | MMC_CAP_POWER_OFF_CARD;
+
+	if (is_mmc_8xxx(host))
+		mmc->caps |= MMC_CAP_3_3V_DDR;
+
+	mmc->caps |= MMC_CAP_ERASE | MMC_CAP_CMD23 | MMC_CAP_POWER_OFF_CARD;
 
 	if (host->use_sg)
 		mmc->max_segs = 16;
-- 
2.17.1

